qutip/qip/circuit.py

```diff
@@ -1147,3 +1147,174 @@ def qasm(self):
             code += "\n"
 
         return code
+        
+    @classmethod
+    def read_qasm(cls, file_path):
+        """
+        Import a QSAM 1.0 file and return an equivalent QuTiP quantum
+        circuit.
+
+        Parameters
+        ----------
+        file_path : string
```

Comment:
Missing `cls` as parameter. Please add `Returns:` with the output.

Reply:
@nathanshammah `cls` is a pseudo-parameter, like `self`.  When you decorate a function with `@classmethod`, then you call it as (say) `QubitCircuit.read_qasm(file_path)`, and `cls` becomes the `type` `QubitCircuit` (as opposed to a specific instance).  It's like a static method in any other OOP language.